sdk/identity/Azure.Identity/src/TokenCredentialOptions.cs

```diff
@@ -21,7 +21,8 @@ public class TokenCredentialOptions : ClientOptions
         /// </summary>
         public TokenCredentialOptions()
         {
-            AuthorityHost = KnownAuthorityHosts.AzureCloud;
+            Uri authorityHostUri = string.IsNullOrEmpty(EnvironmentVariables.AuthorityHost) ? KnownAuthorityHosts.AzureCloud : new Uri(EnvironmentVariables.AuthorityHost);
```

Comment:
Should be robust against the case where `AZURE_AUTHORITY_HOST` is set to some value which can not be parsed as a `Uri`?

Reply:
If we do that for env var, then would make sense to do for the authorityHost prop as well.

Reply:
@ellismg what would you expect the behavior to be in the case where `AZURE_AUTHORITY_HOST` is not a valid Uri? Throw a more specific exception, or ignore the environment variable. If we are going to handle this case I would vote to just raise a more specific exception.

Reply:
Agree.  If there's a value and it's not valid, then they should know.

Reply:
> @ellismg what would you expect the behavior to be in the case where AZURE_AUTHORITY_HOST is not a valid Uri?

My concern here is cases where someone is trying to explicitly set this value, and for some reason the environment variable has garbage data in it. In this case, I'm not even sure if you could create a `TokenCredentialOptions` instance, to be able to set `AuthorityHost` manually.

Reply:
That is a good point. To allow that we would need to reconsider where we read the environment variable and set the default. We could do something like this rather than setting it in the constructor.

~~~ c#
public class TokenCredentialOptions
{
   private Uri _authorityHost;

   public Uri AuthorityHost
   {
      get { return _authorityHost ?? new Uri(EnvironmentVariables.AuthorityHost) ?? KnownAuthorityHosts.AzureCloud; }
      set { _autorityHost = value; }
   }
}
~~~
Thoughts?

Reply:
FWIW - If we were able to delay this until the first time the `AuthorityHost` property was read, I would be fine throwing there.  I think in that case users would still be able to write code like:

````
new TokenCredentialOptions() {
    AuthorityHost = new Uri("https://....")
};
```

Reply:
We're just ships passing in the night, Scott :-).  That makes sense to me.

Reply:
Update the code to read environment variable using get accessor. Error will occur when using a null to new a Uri instance. Hence, I also add a null validation for Authority host variable. So that we can make sure using the default host when not setting environment variable.